### PR TITLE
Fix build tagging when '@' is used

### PR DESCRIPTION
**Pull request by Martchus**
Which other characters would be valid?

### CODE REVIEW

**Comment by okurz** [comment]
LGTM

Fixes https://progress.opensuse.org/issues/11934

**Comment by Martchus** [comment]
I adjusted all tests to work with the new fixture. There are still some tests failing, but I don't think that's related to this commit.

**Comment by okurz** [comment]
LGTM, agreed

**Comment by coolo** [comment]
you will have to rebase to find out

File: t/ui/10-tests_overview.t [hunk 1]
```diff
@@ -96,8 +96,7 @@ like($summary, qr/Passed: 2 Failed: 0 Scheduled: 2 Running: 2 None: 1/i);
 $get = $t->get_ok('/tests/overview' => form => {distri => 'opensuse', version => 'Factory'});
 $get->status_is(200);
 $summary = OpenQA::Test::Case::trim_whitespace($t->tx->res->dom->at('#summary')->all_text);
-like($summary, qr/\QSummary of opensuse Factory build 0048\E/i);
-like($summary, qr/\QPassed: 0 Soft Failure: 1 Failed: 1\E/i);
+is($summary, 'Overall Summary of opensuse Factory build 0048@0815 Passed: 0 Failed: 1');
```
Comment (Martchus) [inline t/ui/10-tests_overview.t]: I don't understand this regex. Maybe someone can explain or we just leave the hard coded version. If something will change, it is very likely that we need to adjust it again anyways.

**Comment by Martchus** [comment]
Haven't noticed that there is a conflict with the current base. I resolved the conflict, but note my comment about the regex.

File: t/ui/10-tests_overview.t (hunk 1 above)
Comment (aaannz) [inline t/ui/10-tests_overview.t]: `\Q .. \E` means perl will ignore regexp meta characters `()[].*`... in between `\Q` and `\E`. `/i` at the end makes regex matching case insensitive.

See http://perldoc.perl.org/perlre.html

Comment (Martchus) [inline t/ui/10-tests_overview.t]: Ok. I just wasn't sure how this is supposed to match anything without the "Soft Failure: 1" like the actual summary. But this change seems to be also cause by adding the new fixture.

**Comment by Martchus** [comment]
I fixed the test again. Seems like the test system used by Travis doesn't like @ inside literal sections of regex. (The test ran fine on my local machine.)

**Comment by okurz** [comment]
did you check why the coverage decreased? You can click on the coverage icon by coveralls and should be able to see where it reduced

File: t/17-build_tagging.t [hunk 2]
```diff
@@ -65,24 +65,31 @@ subtest 'tag icon on group overview on important build' => sub {
     is(scalar @tags, 1,    'one build tagged');
     is($tags[0],     'GM', 'tag description shown');
 };
+subtest 'test whether tags with @ work, too' => sub {
```
Comment (okurz) [inline t/17-build_tagging.t]: please add a separating space in before

Comment (Martchus) [inline t/17-build_tagging.t]: Updated

**Comment by okurz** [comment]
LGTM

File: t/ui/14-dashboard.t [hunk 3]
```diff
@@ -53,7 +53,7 @@ is(scalar @{$driver->find_elements('h4', 'css')}, 2, 'only one build per group s
 
 $driver->find_element('opensuse', 'link_text')->click();
 
-is(scalar @{$driver->find_elements('h4', 'css')}, 3, 'default number of builds shown');
+is(scalar @{$driver->find_elements('h4', 'css')}, 4, 'default number of builds shown');
```
Comment (coolo) [inline t/ui/14-dashboard.t]: This looks fishy. The default number is still 3

Comment (Martchus) [inline t/ui/14-dashboard.t]: Yes, I'll check what's actually wrong with the test.

Comment (Martchus) [inline t/ui/14-dashboard.t]: The default number is 10 (see lib/OpenQA/WebAPI/Controller/Main.pm, line 146) so the increase is ok (the default sets the _maximum_ number of builds).

**Comment by coolo** [comment]
I pushed 914a60ddd24b772aa8dfa496381d11ce9f2ebab7 together with this

**Pull request closed without merging**